app/assets/javascripts/discourse/lib/screen-track.js.es6

```diff
@@ -115,26 +113,52 @@ const ScreenTrack = Ember.Object.extend({
     Discourse.TopicTrackingState.current().updateSeen(topicId, highestSeen);
 
     if (!$.isEmptyObject(newTimings)) {
-      Discourse.ajax('/topics/timings', {
-        data: {
-          timings: newTimings,
-          topic_time: this.get('topicTime'),
-          topic_id: topicId
-        },
-        cache: false,
-        type: 'POST',
-        headers: {
-          'X-SILENCE-LOGGER': 'true'
+      if (Discourse.User.current()) {
+        Discourse.ajax('/topics/timings', {
+          data: {
+            timings: newTimings,
+            topic_time: this.get('topicTime'),
+            topic_id: topicId
+          },
+          cache: false,
+          type: 'POST',
+          headers: {
+            'X-SILENCE-LOGGER': 'true'
+          }
+        }).then(function() {
+          const controller = self.get('topicController');
+          if (controller) {
+            const postNumbers = Object.keys(newTimings).map(function(v) {
+              return parseInt(v, 10);
+            });
+            controller.readPosts(topicId, postNumbers);
+          }
+        });
+      } else {
+        // Anonymous viewer - save to localStorage
+        const store = this.get('keyValueStore');
```

Comment:
I find using the name `store` here quite confusing since the `store` means something totally different in our app. Could you replace with `keyValueStore` or `kvsStore` or something more clear?